general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -920,6 +933,53 @@ def __repr__(self) -> str:
         )
 
 
+def read_ibm_credentials(ibm_name: str) -> dict[str, str]:
+    """Function to try to read IBM credentials from .qiskit/qiskit-ibm.json.
+
+    Args:
+        ibm_name: The name under which the IBM account credentials are locally stored.
+
+    Raises:
+        FileNotFoundError: If the file is not found.
+        KeyError: If the provided `ibm_name` does not have credentials stored in the
+            config file or `token` and/or `channel` keys are missing for the credentials
+            of the account under `ibm_name`.
+
+    Returns:
+        Dictionary containing the ibm token, channel, and instance (if available).
+    """
+    data_dir = pathlib.Path(os.getenv("XDG_DATA_HOME", "~/.local/share")).expanduser()
+    home_dir = pathlib.Path.home()
+    for directory in [
+        home_dir.joinpath(".qiskit"),
+        data_dir.joinpath("super.tech"),
+        data_dir.joinpath("coldquanta"),
+        home_dir.joinpath(".super.tech"),
+        home_dir.joinpath(".coldquanta"),
+    ]:
+        path = directory.joinpath("qiskit-ibm.json")
+        if path.is_file():
+            config = json.load(open(path))
+            if ibm_name not in config:
```

Comment:
if the user doesn't pass in a name we should handle it closer to what qiskit does instead of assuming a particular name. i think this means:
1. look for any account they've marked as default (by checking `credentials.get("is_default_account")`)
2. look for accounts named `["default-ibm-cloud", "default-ibm-quantum", "default"]` (in that order i think)
3. look for the last account with the channel "ibm-cloud"
4. return the last account in the config

Reply:
alternatively, if we're not given a name maybe we should just check for `credentials.get("is_default_account")`, and if that doesn't work throw an error to avoid doing something the user might not expect (unless there is exactly one account in the config, in which case we should just return that)